### PR TITLE
path exclusion in no longer os-dependent

### LINKED ISSUE #51: [BUG] Exclude paths not os independent
### Environment

- `interrogate` version(s) (`interrogate --version`: 1.3.1
- Operating System(s): Windows10
- Python version(s): 3.7.8

### Description of the bug
In one of my projects I exclude a subdir with autogenerated files, form being tested with `inrerrogate`.
In order for this to work crossplatform I need to define the exclusion twice.
Once with windows path separators `\\` and once with Posix separators `/`.
So I end up with the following config:
```toml
[tool.interrogate]
exclude = ["setup.py", "docs", "tests", ".eggs", "djangocms_equation\\migrations", "djangocms_equation/migrations"]
```

### What you expected to happen
Just using `"djangocms_equation/migrations"` should be sufficient.

### How to reproduce (as minimally and precisely as possible)
Just exclude a file with posix path noatation on windows, and it won't be excluded.

### Anthing else we need to know?
This can be fixed by using `os.path.normpath` in the following line.
https://github.com/econchick/interrogate/blob/624dc095d44aeaf34c73f29151111a14588eb129/src/interrogate/coverage.py#L131-L136

```python 
            maybe_excluded_dir = any(
                [f.startswith(os.path.normpath(exc)) for exc in self.excluded]
            )
```

Also `maybe_excluded_file ` seams to be redundant since `some_str.startswith(some_str) == some_str in (some_str,)`

```python
In [1]: file_path = "/home/user/foo"

In [2]: exclude = (file_path, )

In [3]: any(file_path.startswith(exc) for exc in exclude)
Out[3]: True

In [4]: file_path in exclude
Out[4]: True
```

### PR DESCRIPTION
**Pull request by oriash93**
# Hey, I just made a Pull Request!

## Description
Excluding path is no longer os-dependent, by using `os.path.normpath`.

## Motivation and Context
Fixes #51 

## Have you tested this? If so, how?
I ran `interrogate` with these changes over relevant cases and it works for me.
No additional unit tests required.

## Checklist for PR author(s)

- [x] Changes are covered by unit tests (no major decrease in code coverage %).
- [X] All tests pass.
- [X] Docstring coverage is **100%** via `tox -e docs` or `interrogate -c pyproject.toml` (I mean, we _should_ set a good example :smile:).
- [X] Updates to documentation:
    - [X] Document any relevant additions/changes in `README.rst`.
    - [X] Manually update **both** the `README.rst` _and_ `docs/index.rst` for any new/changed CLI flags.
    - [X] Any changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).  Find the appropriate next version in the project's [``__init__.py``](https://github.com/econchick/interrogate/blob/master/src/interrogate/__init__.py) file.

### CODE REVIEW

**Comment by s-weigand** [comment]
Someone didn't [install the `pre-commit` hook](https://github.com/econchick/interrogate/blob/master/CONTRIBUTING.md) 😉

File: src/interrogate/coverage.py [hunk 1]
```diff
@@ -128,11 +128,7 @@ def _filter_files(self, files):
                 basename = os.path.basename(f)
                 if basename == "__init__.py":
                     continue
-            maybe_excluded_dir = any(
-                [f.startswith(exc) for exc in self.excluded]
-            )
-            maybe_excluded_file = f in self.excluded
-            if any([maybe_excluded_dir, maybe_excluded_file]):
+            if any([f.startswith(os.path.normpath(exc)) for exc in self.excluded]):
```
Comment (econchick) [inline src/interrogate/coverage.py:131]: hey @oriash93 - thanks for the PR! Glad it was just a small change :) It looks like formatting is failing the build. Can you run `tox -e lint` locally to take care of that, then push your changes up?

**Review by econchick — CHANGES_REQUESTED** [review]

**Comment by s-weigand** [comment]
@oriash93 if @econchick is fine with the proposed feature in #59 , could you change
```python
if any(
                [f.startswith(os.path.normpath(exc)) for exc in self.excluded]
            ):
```
to

```python
if any( fnmatch(f, exc) for exc in self.excluded ):
```

That way this PR could kill two birds with one stone.

**Comment by oriash93** [comment]
Whatever @econchick decides.

Also, on a different note - do you mind marking this PR or repo as eligible for hacktoberfest?

**Comment by s-weigand** [comment]
Thinking about it, the replacement should be
```python
if any( fnmatch(f, exc + "*") for exc in self.excluded ):
```
or if the python 3.5 support was droped before
```python
if any( fnmatch(f, f"{exc}*") for exc in self.excluded ):
```

That way it won't break configs that work with the current `startswith` implementation.

```python
In [1]: fnmatch(r"project\tests\test_file.py", "project/tests")
Out[1]: False

In [2]: fnmatch(r"project\tests\test_file.py", "project/tests*")
Out[2]: True

In [3]: fnmatch(r"project\tests\test_file.py", "project/tests**")
Out[3]: True

In [4]: fnmatch(r"project\tests\test_file.py", "project/tests***")
Out[4]: True

In [5]: fnmatch(r"project\tests\test_file.py", "project/tests****")
Out[5]: True
```

**Comment by econchick** [comment]
Thanks @oriash93 @s-weigand (btw I labeled the repo w/ `hacktoberfest` - will that be enough?)

I'm cool with `fnmatch` if it makes #59 easier, although I'm not sure it does. I guess it depends on what's wanted: if we want to support _regexes_, then `fnmatch` may not be the best way since it [does not support `re` regex](https://docs.python.org/3/library/fnmatch.html). If #59 just wants to support unix-style wildcards, then it should be enough. That said, I'm not that familiar with `fnmatch` myself, so I could be wrong :)

**Comment by s-weigand** [comment]
`fnmatch` uses unix-style glob patterns, which I guess most people are more familiar with than RegEx.
At least I hear all the time "I'm scared of RegEx" or "I never understood RegEx" 😋 
Also `fnmatch` is the way other tools like `flake8` handle their file matching.

In #59 I was just asking for a less explicit/ more general way to exclude files, so glob patterns are totaly awesome for me. 😄

**Comment by oriash93** [comment]
pinging @econchick

**Comment by econchick** [comment]
Thanks so much! Sorry for the delay - I was on vacation for the past week and a half 😄

**Comment by econchick** [comment]
FYI this has been released (finally) in [1.3.2](https://pypi.org/project/interrogate/1.3.2/) 🎉  - thanks again for you're help @oriash93 & @s-weigand 🥇

**Pull request merged**